### ISSUE TITLE
Incorrect centroid calculated for non-convex quads

### ISSUE THREAD

**Issue by dmac000** [NREL/EnergyPlus#5497]
When creating a small non-convex building surface, then I get the error:

```
** Severe  ** SetOutBulbTempAt: Surface Outdoor Temperatures < -100 C
**   ~~~   ** ...check Surface Heights - Maximum Surface Height=[110465].
**   ~~~   ** ...according to your maximum Z height, your building is somewhere in the Stratosphere.
**  Fatal  ** Program terminates due to preceding condition(s).
```

with the vertices:

```
9.857091, -27.472747, 11.0,  !- Vertex 1 position
5.065178, -24.431707, 11.0,  !- Vertex 2 position
9.857181, -27.471673, 11.0,  !- Vertex 3 position
14.461132, -20.217025, 11.0; !- Vertex 4 position
```

It looks like the high z-value is coming from `CalcSurfaceCentroid` in `SurfaceGeometry.cc` where 4-sided polygons are split into 3-sided polygons. This split is not correct for non-convex surfaces. The value is calculated as:

```
TotalArea = 0.00485596
Tri1Area = 24.380 / TotalArea = 5020.642
Tri2Area = 24.384 / TotalArea = 5021.642

Tri1Area * 11 + Tri2Area * 11 = 110465
```

**Comment by mjwitte** [comment]
If I'm following the vertices correctly, this is not a well-formed polygon, the sides cross over each other.  Seems that vertices 3 and 4 are reversed?  Also, vertex 3 is nearly on top of vertex1 and the x and y differences between these two are less than the tolerance for geometry.  From the I/O Ref section titled "Surface Vertices":

> Note that the resolution on the surface vertex input is 1 millimeter (.001 meter). Therefore, using vertices that are very close together (<1 mm) may result in invalid dot product and fatal errors during shading calculations.

Were there any other warning messages about this surface in the error file prior to the fatal?

**Comment by dmac000** [comment]
The surface is a v-shape which should be well-formed. The extreme example produces a user-visible error because of the 'stratosphere' check. This is the only error that is shown. More moderate examples produce the wrong values but are not enough to produce an error.

Another example is something like (10, 10, 0), (20, 50, 0), (30, 10, 0), (20, 60, 0). As points 2 and 4 get closer together, the error in the centroid increases because it's forming a triangle using the first 3 points which include the area outside the polygon. Because the area of the triangles over the area of the polygon doesn't add up to 1, then the center point becomes a multiple of what it should be.

**Comment by mjwitte** [comment]
My hand-plot of the original points was flawed.  Yes, it's a v-shape but very severe.
![image](https://cloud.githubusercontent.com/assets/4991075/13185278/dce1cb20-d705-11e5-9bd1-a9747abd143f.png)
![image](https://cloud.githubusercontent.com/assets/4991075/13185311/05c57226-d706-11e5-95b3-b8520304c99a.png)

The second set of points are more reasonable.
![image](https://cloud.githubusercontent.com/assets/4991075/13185350/42e4af96-d706-11e5-8060-faa3ea596359.png)

In either case, however, since all four vertices have the same z-value, one would expect at least the z coordinate of the centroid to be correct.  If you have working idf files illustrating both of these, please attach them here.

**Comment by dmac000** [comment]
I can't distribute the original files but I've uploaded a similar idf showing the same error:
[idfinput.zip](https://github.com/NREL/EnergyPlus/files/140517/idfinput.zip)

**Comment by JasonGlazer** [comment]
Here is the input file updated to the prerelease version of 8.6.0 [5497-idfinput-V860.zip](https://github.com/NREL/EnergyPlus/files/442989/5497-idfinput-V860.zip)

**Comment by mjwitte** [comment]
Closed via #5852
